featuretools/tests/entityset_tests/test_es.py

```diff
@@ -44,72 +47,164 @@ def test_cannot_re_add_relationships_that_already_exists(es):
+        'product_id': ltypes.Categorical,
+        'datetime': ltypes.Datetime,
+        'value': ltypes.Double,
+        'value_2': ltypes.Double,
+        'latlong': ltypes.LatLong,
+        'latlong2': ltypes.LatLong,
+        'zipcode': ltypes.PostalCode,
+        'countrycode': ltypes.CountryCode,
+        'subregioncode': ltypes.SubRegionCode,
+        'value_many_nans': ltypes.Double,
+        'priority_level': ltypes.Ordinal(order=[0, 1, 2]),
+        'purchased': ltypes.Boolean,
+        'comments': ltypes.NaturalLanguage
+    }
+    log_semantic_tags = {
+        'session_id': 'foreign_key',
+        'product_id': 'foreign_key'
+    }
+    assert set(log_logical_types) == set(log_2_df.columns)
+    es.add_dataframe(dataframe_id='log2',
+                     dataframe=log_2_df,
+                     index='id',
+                     logical_types=log_logical_types,
+                     semantic_tags=log_semantic_tags,
+                     time_index='datetime')
+    assert 'log2' in es.dataframe_dict
+    assert es['log2'].ww.schema is not None
+    assert es['log2'].ww.logical_types['session_id'] == ltypes.Datetime
+    assert es['customers'].ww.logical_types['id'] == ltypes.Integer
+
+    warning_text = f'Logical type Datetime for child column session_id does not match parent column id logical type Integer. Changing child logical type to match parent.'
+    with pytest.warns(UserWarning, match=warning_text):
         es.add_relationship(u'customers', 'id', 'log2', 'session_id')
+    assert es['log2'].ww.logical_types['session_id'] == ltypes.Integer
+    assert es['customers'].ww.logical_types['id'] == ltypes.Integer
 
 
 def test_add_relationship_errors_child_v_index(es):
-    log_df = es['log'].df.copy()
-    log_vtypes = es['log'].variable_types
-    es.entity_from_dataframe(entity_id='log2',
-                             dataframe=log_df,
-                             index='id',
-                             variable_types=log_vtypes,
-                             time_index='datetime')
+    es.add_dataframe(dataframe_id='log2',
+                     dataframe=es['log'].ww.copy())
 
     to_match = "Unable to add relationship because child column 'id' in 'log2' is also its index"
     with pytest.raises(ValueError, match=to_match):
         es.add_relationship('log', 'id', 'log2', 'id')
 
 
-def test_add_relationship_empty_child_convert_dtype(es):
-    relationship = ft.Relationship(es, "sessions", "id", "log", "session_id")
-    es['log'].df = pd.DataFrame(columns=es['log'].df.columns)
-    assert len(es['log'].df) == 0
-    assert es['log'].df['session_id'].dtype == 'object'
+# def test_add_relationship_empty_child_convert_dtype(es):
+#     relationship = ft.Relationship(es, "sessions", "id", "log", "session_id")
+#     empty_log_df = pd.DataFrame(columns=es['log'].columns)
+#     # --> schema isn't valid.... not sure what this is testing
+#     empty_log_df.ww.init(schema=es['log'].ww.schema)
```

Comment:
I think this was testing that if you add a relationship between two columns, that the dtype of the child column gets updated to match the dtype of the parent columns, a condition which was enforced. This test seems to be related to  the entityset code block you highlighted above where the child column dtype was set to `int64` if the dataframe was empty.

I think the equivalent test with WW would be to create an entityset with one empty dataframe (the child), with the child column in the relationship having a different ltype/dtype than the parent column, and no relationships defined initially. Then add a relationship between those columns and verify that the logical types of the columns (and thus the dtypes) in the relationship are the same. This will probably also raise a warning now, if I'm following along correctly.

Reply:
ahh got it. I feel like that's more of a test of the logical type change warning in general, which we have in a couple of places in `test_es`, and the Woodwork DataFrames with no data is maybe something we should be testing in Woodwork, but I think that maybe this test and the lines in entityset where we convert to `int64` are going to be covered by the logical type change, or, like you said the fact that if the logical types are the same, then the dtypes should be the same.